includes/classes/Indexable/Post/Post.php

```diff
@@ -1625,4 +1625,88 @@ protected function get_orderby_array( $orderbys ) {
 
 		return $orderbys;
 	}
+
+	/**
+	 * Check if a value matches a protected parameter
+	 *
+	 * @see https://developer.wordpress.org/reference/classes/wp_query/#parameters List of protected parameters
+	 * 
+	 * @param string $value The value to test
+	 * @since 3.4
+	 * @return bool
+	 */
+	private function is_protected_parameter( $value ) {
+		if( ! is_string( $value ) ) {
+			return false;
+		}
+		
+		$protected_parameter_list = array(
+			'author',
+			'author__in',
+			'author__not_in',
+			'author_id',
+			'author_name',
+			'cache_results',
+			'cat',
+			'category__and',
+			'category__in',
+			'category__not_in',
+			'category_name',
+			'comment_count',
+			'date_query',
+			'day',
+			'fields',
+			'has_password',
+			'hour',
+			'ignore_sticky_posts',
+			'm',
+			'meta_compare',
+			'meta_key',
+			'meta_query',
+			'meta_value',
+			'meta_value_num',
+			'minute',
+			'monthnum',
+			'name',
+			'nopaging',
+			'offset',
+			'order',
+			'orderby',
+			'p',
+			'page',
+			'page_id',
+			'paged',
+			'pagename',
+			'perm',
+			'post__in',
+			'post__not_in',
+			'post_mime_type',
+			'post_name__in',
+			'post_parent',
+			'post_parent__in',
+			'post_parent__not_in',
+			'post_password',
+			'post_status',
+			'post_type',
+			'posts_per_archive_page',
+			'posts_per_page',
+			's',
+			'second',
+			'tag',
+			'tag__and',
+			'tag__in',
+			'tag__not_in',
+			'tag_id',
+			'tag_slug__and',
+			'tag_slug__in',
+			'tax',
+			'tax_query',
+			'update_post_meta_cache',
+			'update_post_term_cache',
+			'w',
+			'year',
+		);
+
+		return in_array( $value, $protected_parameter_list );
```

Comment:
```suggestion
		return in_array( $value, $protected_parameter_list, true );
```

Minor - it's usually best to use the "strict" mode of in_array() by passing `true` as the 3rd param

Reply:
With the string type check, only strings will make it that far right now.

However, future proofing is a good idea and you already typed it out :+1:.